File: spec/functional_test/fixtures/php/laravel_auth/app/Http/Controllers/PostController.php
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;

class PostController extends Controller
{
    public function __construct()
    {
        $this->middleware('auth');
    }

    public function index()
    {
        return response()->json(Post::all());
    }

    public function store(Request $request)
    {
        $this->authorize('create', Post::class);
        return response()->json(Post::create($request->all()));
    }
}

File: spec/functional_test/fixtures/python/masonite/app/controllers/PostController.py
"""A PostController Module — implements every action Route.resource /
Route.api generate so the fixture's resource routes actually resolve."""
from masonite.controllers import Controller
from masonite.request import Request
from masonite.response import Response


class PostController(Controller):
    def index(self, response: Response):
        return response.json([])

    def create(self, response: Response):
        return response.view("posts.create")

    def store(self, request: Request, response: Response):
        title = request.input("title")
        return response.json({"title": title})

    def show(self, response: Response, id):
        return response.json({"id": id})

    def edit(self, response: Response, id):
        return response.json({"id": id})

    def update(self, request: Request, response: Response, id):
        title = request.input("title")
        return response.json({"id": id, "title": title})

    def destroy(self, response: Response, id):
        return response.json({"id": id})
